src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,40 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int EIGHTEEN_YEARS_OLD = 18;
+    private static final int MIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new NullPointerException("User can't be Null");
+        }
+        if (user.getLogin() == null) {
+            throw new NullPointerException("Login is null");
+        }
+        if (user.getAge() == null) {
+            throw new NullPointerException("Age is null");
+        }
+        if (user.getPassword() == null) {
+            throw new NullPointerException("Password is null");
+        }
+        for (User u : Storage.people) {
```

Comment:
```suggestion
        for (User user : Storage.people) {
```
User storageDao methods instead